cmd/kops/validate_cluster.go

```diff
@@ -205,96 +190,52 @@ func validateClusterOutputTable(validationCluster *validation.ValidationCluster,
-		// TODO: Maybe print the instance group role instead?
-		// TODO: Maybe include the instance group name?
-		role := apiutil.GetNodeRole(&n)
-		if role == "" {
-			role = "node"
-		}
-		return role
-	})
-
-	fmt.Fprintln(out, "\nNODE STATUS")
-	err = nodeTable.Render(validationCluster.NodeList.Items, out, "NAME", "ROLE", "READY")
-
-	if err != nil {
-		return fmt.Errorf("cannot render nodes for %q: %v", validationCluster.ClusterName, err)
-	}
+	{
+		nodeTable := &tables.Table{}
+		nodeTable.AddColumn("NAME", func(n *validation.ValidationNode) string {
+			return n.Name
+		})
 
-	if len(validationCluster.ComponentFailures) != 0 {
-		componentFailuresTable := &tables.Table{}
-		componentFailuresTable.AddColumn("NAME", func(s string) string {
-			return s
+		nodeTable.AddColumn("READY", func(n *validation.ValidationNode) v1.ConditionStatus {
+			return n.Status
 		})
 
-		fmt.Fprintln(out, "\nComponent Failures")
-		err = componentFailuresTable.Render(validationCluster.ComponentFailures, out, "NAME")
+		nodeTable.AddColumn("ROLE", func(n *validation.ValidationNode) string {
+			return n.Role
+		})
 
-		if err != nil {
-			return fmt.Errorf("cannot render components for %q: %v", validationCluster.ClusterName, err)
+		fmt.Fprintln(out, "\nNODE STATUS")
+		if err := nodeTable.Render(result.Nodes, out, "NAME", "ROLE", "READY"); err != nil {
+			return fmt.Errorf("cannot render nodes for %q: %v", cluster.Name, err)
 		}
 	}
 
-	if len(validationCluster.PodFailures) != 0 {
-		podFailuresTable := &tables.Table{}
-		podFailuresTable.AddColumn("NAME", func(s string) string {
-			return s
+	if len(result.Failures) != 0 {
+		failuresTable := &tables.Table{}
+		failuresTable.AddColumn("KIND", func(e *validation.ValidationError) string {
+			return e.Kind
+		})
+		failuresTable.AddColumn("NAME", func(e *validation.ValidationError) string {
+			return e.Name
+		})
+		failuresTable.AddColumn("MESSAGE", func(e *validation.ValidationError) string {
+			return e.Message
 		})
 
-		fmt.Fprintln(out, "\nPod Failures in kube-system")
-		err = podFailuresTable.Render(validationCluster.PodFailures, out, "NAME")
-
-		if err != nil {
-			return fmt.Errorf("cannot render pods for %q: %v", validationCluster.ClusterName, err)
+		fmt.Fprintln(out, "\nVALIDATION ERRORS")
+		if err := failuresTable.Render(result.Failures, out, "KIND", "NAME", "MESSAGE"); err != nil {
+			return fmt.Errorf("error rendering failures table: %v", err)
 		}
 	}
 
-	if validationFailed == nil {
-		fmt.Fprintf(out, "\nYour cluster %s is ready\n", validationCluster.ClusterName)
-		return nil
+	if len(result.Failures) == 0 {
+		fmt.Fprintf(out, "\nYour cluster %s is ready\n", cluster.Name)
 	} else {
-		// do we need to print which instance group is not ready?
-		// nodes are going to be a pain
 		fmt.Fprint(out, "\nValidation Failed\n")
-		fmt.Fprintf(out, "Ready Master(s) %d out of %d.\n", len(validationCluster.MastersReadyArray), validationCluster.MastersCount)
```

Comment:
Where did you move this output to? Again seems to be code improvements / UX changes not dealing with the bug.

Reply:
Should we have these changes in another PR?

Reply:
It's in the failure list.  The problem is that our model - particularly for nodes - is richer than a simple count, so these messages would do things like "Ready nodes 10 out of 6" etc.

Reply:
Frankly as an operator that part of the ux was incredibly useful. I would prefer to not remove it

Reply:
Yes - I would like to see it in the instance group grid above.  It just isn't well-defined here, sadly.